### DISCUSSION TITLE
TreeSelect component checkboxes behavior

### DISCUSSION THREAD

**Discussion by rdyava** [primefaces/community#1298] [PrimeReact] [answered]
Hello all,

I am using the TreeSelect component and was wondering if there is a way to alter the behavior of the checkbox selection mode. Currently when there is a single child node of a parent node, with the checkbox selection mode enabled and selecting the child, it also checks the parent node as well. Is there a way to make it so that for this sort of use case, the parent node is not selected? For example, in the example in the documentation, selecting "Invoices.txt" also selects "Home", I would need "Home" to be a partially selected value in this case and not showing as a selection with "Invoices.txt". 

![image](https://github.com/primefaces/community/assets/47518433/c3019b80-a732-41d8-a33a-d7fa7f7a3f77)

**Comment by melloware** [comment, marked as answer]
Yes see these properties in the docs...

https://primereact.org/tree/#api.Tree.props.propagateSelectionUp

https://primereact.org/tree/#api.Tree.props.propagateSelectionDown

**Reply by JL45312** [reply]
Does propagateSelectionUp/Down work with TreeSelect? It is not listed as a prop in https://primereact.org/treeselect/#api.TreeSelect.props

**Reply by melloware** [reply]
You don't need that right treeSelect only selects a single item right?

**Reply by rdyava** [reply]
The checkbox selection mode allows for multiple selections with TreeSelect, which is a use case I need to solve this issue for if it’s possible with this component. The props you linked would work for my needs, but like the other comment mentioned, it’s for Tree not TreeSelect.

**Reply by melloware** [reply]
Ok can you open a github ticket requesting that feature please.
